src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
         unimplemented!()
     }
 
+    fn quoted_arg(&self) -> bool {
+        unimplemented!()
+    }
+
+    fn request_or_macro(&mut self) -> bool {
+        // self.skip_char();
+        self.s = self.skip_char(&self.s, None).to_owned();
+
+        let s0 = &self.s[1..2];
```

Comment:
Possible out of bound error here, I believe we should use `self.s.bytes().get(1)` or something safer here.

Reply:
taken care of:
```rust
let s0 = self.s.chars().nth(1).unwrap_or('_'); // _ will be ignored by the match
```

note: I changed the type of the match from `&str` to `char` because it's only one character

Reply:
Wait...is this actually resolved? Did you add the code you mention somewhere else?

Reply:
Actually fixed now